src/_output/vue/app.vue

```diff
@@ -66,7 +74,19 @@ export default {
         { value: 'viewSentence', label: 'Sentence'}
       ],
       viewType: 'viewFull',
-      sentenceCount: 0
+      sentenceCount: 0,
+      languagesList: []
+    }
+  },
+  created() {
+    this.languagesList = GroupUtility.allLanguagesTargets(this.fullData)
+  },
+  computed: {
+    fullData () {
+      return this.$parent.fullData
+    },
+    languageTargetIds () {
+      return this.languagesList.filter(langData => !langData.hidden).map(langData => langData.targetId)

```

Comment:
There are several way to organize code in the Vue component between `watch`, `computed` and `methods`

`watch` - executes on each watched property change (I use only for simple types)
`computed` - executes only when any observed property inside was changed, otherwise it is cached
`methods` - executes each time